src/player.cpp

```diff
@@ -3550,6 +3550,12 @@ bool player::wield_contents( item &container, item *internal_item, bool penaltie
         inv.unsort();
     }
 
+    // for holsters, we should not include the cost of wielding the holster itself
+    // It looks like the cost of wielding the container is removed earlier, but I can't found where.
+    // As such, I'm re-adding it here.
```

Comment:
Ideally we resolve these comments before the PR is merged.

Reply:
Alright, I'm digging into it.

Firstly, I'm considering the Moves value in the panel at the top right corner as authoritative. It uses ``movecounter``, which is the difference between the number of moves before and after the action.
It seems the values of ``movecounter`` and ``mv``, the parameter of the item::on_wield function which choose the sentence "It takes you [several seconds/a long time/...] to wield your <object>.", weren't consistent.

I dumped the full stacktrace for wielding a weapon by dr[a]wing from a holster, using GDB 9.2:
- item::on_wield(p,mv) at item.cpp:4357
- player::wield_contents(container,internal_item,penalties,base_cost) at player.cpp:3571
- holster_actor::use(p,it) at iuse_actor.cpp:2420
- itype::invoke(p,it,pos,iuse_name) at itype.cpp:117
- Character::invoke_item(used,method,pt)  at character.cpp:7890
- avatar::invoke_item(used,pt) at avatar.cpp:1586
- player::use(loc) at player.cpp:2991
- avatar_action::use_item(you,loc) at avatar_action.cpp:1099
- avatar_action::use_item(you) at avatar_action.cpp:1096
- game::handle_action() at handle_action.cpp:2012
- game::do_turn() at game.cpp:1470
- main(argc,argv) at main.cpp:506

It seems that the cost of the holster is added in avatar_action::use_item(you,loc), where it calls loc.obtain on the holster.
In fact, this additional cost can be avoided if the item has the ALLOWS_REMOTE_USE flag.

We then have three solutions:
- We add the ALLOWS_REMOTE_USE flag to every holsters. The problem is, if an holster doesn't have that flag, there will be an inconsistency between [w] and [a].
- We "whitelist" the holsters, transforming the ``loc->has_flag( flag_ALLOWS_REMOTE_USE`` condition to ``loc->has_flag( flag_ALLOWS_REMOTE_USE || loc->get_use( "holster" )``.
The problem is, in the avatar_action::use_item function, we still don't know which action the player will use. If an item has multiple action including holster, the cost of the item may be irrelevantly removed when the used action is not holster.
- We stick to the current situation, though it is not pretty and and might be confusing. I could at least modify the comments to specify where the cost of the holster is added, and why we must remove it here.